src/freenet/client/async/SplitFileInserter.java

```diff
@@ -234,6 +237,11 @@ public boolean run(ClientContext context) {
         });
     }
 
+    protected void unregisterSender() {
+        if (sender != null)
```

Comment:
Why have this check here? I don't see a code path that can result in `sender` being null, which the check implies could happen. If this check is necessary it needs locking to avoid a race condition.